src/wp-includes/class-wp-theme-json.php

```diff
@@ -2128,9 +2156,73 @@ protected static function compute_style_properties( $styles, $settings = array()
 			}
 		}
 
+		if ( $can_use_cached ) {
+			self::set_compute_style_properties_cache( $cache_key, $declarations );
+		}
+
 		return $declarations;
 	}
 
+	/**
+	 * Sets the cache for the compute style properties.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param string $cache_key The cache key.
+	 * @param array  $declarations The declarations to cache.
+	 *
+	 * @return array The cached declarations.
+	 */
+	private static function set_compute_style_properties_cache( $cache_key, $declarations ) {
+		self::$cached_compute_style_properties[ $cache_key ] = $declarations;
+		set_site_transient( 'wp_compute_style_properties', self::$cached_compute_style_properties, HOUR_IN_SECONDS );
+		self::$cached_compute_style_properties;
+	}
+
+	/**
+	 * Gets the cache for the compute style properties.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @return array The cached declarations.
+	 */
+	private function get_compute_style_properties_cache() {
+		self::$cached_compute_style_properties = get_site_transient( 'wp_compute_style_properties' );
+		if ( ! is_array( self::$cached_compute_style_properties ) ) {
+			self::$cached_compute_style_properties = array();
+		}
+	}
+
+	/**
+	 * Clears the cache for the compute style properties.
+	 *
+	 * @since 6.6.0
+	 */
+	public static function clear_compute_style_properties_cache() {
```

Comment:
There is a known bug here. The current code doesn't regenerate `compute_style_properties_cache` after clearing.